### PR TITLE
Wired in StoreKeyConverterFactory

### PR DESCRIPTION
**Pull request by dharju**
Wiring in StoreKeyConverterFactory into AmbryServer, AmbryRequests.  Foundational for merging the converter with other stuff.

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java [hunk 1]
```diff
@@ -61,6 +61,13 @@
   @Default("false")
   public final boolean serverEnableStoreDataPrefetch;
 
+  /**
+   * Implementation class for StoreKeyConverterFactory
+   */
+  @Config("server.store.key.converter.factory.class")
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java]: minor: we (mostly) don't use `.class` at the end for factory config names. (`server.store.key.converter.factory`)

File: ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java [hunk 2]
```diff
@@ -155,11 +157,13 @@ public void startup() throws InstantiationException {
...
       }
 
+      storeKeyConverterFactory = Utils.getObj(serverConfig.serverStoreKeyConverterFactoryClass, properties, registry);
+
       networkServer = new SocketServer(networkConfig, sslConfig, registry, ports);
       requests =
           new AmbryRequests(storageManager, networkServer.getRequestResponseChannel(), clusterMap, nodeId, registry,
               findTokenFactory, notificationSystem, replicationManager, storeKeyFactory,
-              serverConfig.serverEnableStoreDataPrefetch);
+              serverConfig.serverEnableStoreDataPrefetch, storeKeyConverterFactory);
```
Comment (vgkholla) [inline ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java]: `ReplicationManager` needs the `storeKeyConverterFactory` too right? any other components?

File: ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java (hunk 1 above)
Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java]: Got rid of the ".class"

File: ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java (hunk 2 above)
Comment (dharju) [reply ambry-server/src/main/java/com.github.ambry.server/AmbryServer.java]: Added.  I don't believe any others.

File: ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java [hunk 3]
```diff
@@ -61,6 +61,13 @@
   @Default("false")
   public final boolean serverEnableStoreDataPrefetch;
 
+  /**
+   * Implementation class for StoreKeyConverterFactory
+   */
+  @Config("server.store.key.converter.factory")
+  @Default("com.github.ambry.server.StoreKeyConverterFactoryImpl")
+  public final String serverStoreKeyConverterFactoryClass;
```
Comment (pnarayanan) [inline ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java]: the variable has a `Class` at the end that you probably want to remove.

Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/config/ServerConfig.java]: fixed

**Pull request merged**